src/signer_node/message_processor/process_blockvss.rs

```diff
@@ -61,16 +62,17 @@ where
                 params,
             );
 
-            match shared_keys {
-                Some(keys) => NodeState::Master {
+            match result {
+                Err(_) => prev_state.clone(),
+                Ok(Some(keys)) => NodeState::Master {
                     block_key: block_key.clone(),
                     shared_block_secrets: new_shared_block_secrets,
                     block_shared_keys: Some((keys.0, keys.1.x_i, keys.1.y)),
                     candidate_block: candidate_block.clone(),
                     signatures: signatures.clone(),
                     round_is_done: false,
                 },
-                None => NodeState::Master {
+                Ok(_) => NodeState::Master {
```

Comment:
```suggestion
                Ok(None) => NodeState::Master {
```
I prefer this because `_` should be `None`.